modes.go

```diff
@@ -247,9 +248,9 @@ func GenerateData(pk, ck, size int64) []byte {
 				}
 			}
 		} else {
-			reader := rand.ChaCha8{}
 			payload := make([]byte, size-generatedDataHeaderSize-sha256.Size)
-			_, _ = reader.Read(payload)
+			//nolint
+			_, _ = mathrandv1.Read(payload)
```

Comment:
Why not `math/rand/v2.Rand`?

We don't really need `ChaCha8` here even if it hadn't the bug leading to panic.

Reply:
With new random package, the only way to read sized payload, is to use `ChaCha8`, others, like `PCG`, are for `int`s and `float`s, until we figure out how to properly initialize the ChaCha8 with a good seed, in my opinion it's better to use old `math/rand`

Reply:
```
$ git diff
diff --git a/modes.go b/modes.go
index d567978..9f054dc 100644
--- a/modes.go
+++ b/modes.go
@@ -247,7 +247,8 @@ func GenerateData(pk, ck, size int64) []byte {
                                }
                        }
                } else {
-                       reader := rand.ChaCha8{}
+                       seed := sha256.Sum256([]byte(fmt.Sprintf("%d", rand.Uint64())))
+                       reader := rand.NewChaCha8(seed)
                        payload := make([]byte, size-generatedDataHeaderSize-sha256.Size)
                        _, _ = reader.Read(payload)
                        csum := sha256.Sum256(payload)
```

Better not to stick to deprecated libraries.
It is postponing the problem.